lib/manageiq/providers/workflows/runner.rb

```diff
@@ -34,12 +34,12 @@ def stop
           $workflows_log.debug("Runner: Stopping workflows runner...Complete")
         end
 
-        def add_workflow(workflow, queue_args)
-          workflows[workflow.id] ||= [workflow, queue_args]
+        def add_workflow_instance(workflow_instance, queue_args)
+          workflow_instances[workflow_instance.manager_ref] ||= [workflow_instance, queue_args]
```

Comment:
Not sure if this matters but does `||=` work the way you expect with a concurrent Hash?  (or does it matter here)

I thought `x ||= y` was syntactic sugar for `x || x = y`, meaning this would be a check and then a set and would be non-atomic.